src/App.tsx

```diff
@@ -1,8 +1,43 @@
+import { useState } from "react";
+
+import { DisplayCard } from "./components/DisplayCard/DisplayCard";
+
 const App = () => {
+	const [nameInputValue, setNameInputValue] = useState<string>("");
+	const [entities, setEntities] = useState<any[]>([]);
+
+	const queryString = nameInputValue.toLowerCase();
+	const fetchURL = `https://pokeapi.co/api/v2/pokemon/${queryString}/`;
+
+	const fetchEntity = async () => {
+		const response = await fetch(fetchURL);
+		const data = await response.json();
+		setEntities([...entities, data]);
+		setNameInputValue("");
+	};
+
 	return (
 		<>
 			<div>
-				<h1>Pokémon Battle</h1>
+				<input
+					value={nameInputValue}
+					type="text"
+					placeholder="Type a name..."
+					onKeyDown={(e) => {
+						if (e.key === "Enter") {
+							fetchEntity();
+						}
+					}}
+					onChange={(e) => {
+						setNameInputValue(e.target.value);
```

Comment:
this is a conversation we can have about the codebase. My personal opinion is that inline anonymous functions are messy, and cause problems as the codebase grows. idealy i would define this as

```ts
const handleNameInputChange = useCallback((event) => {
    setNameInputValue(event.currentTarget.value);
}. []);
```

and then your onChange  becomes  `onChange={handleNameChangeInput}`

Reply:
Looks like event is deprecated. I'll look into the replacement.

Reply:
I also agree that the event handlers should call defined functions, rather than anonymous inline ones. 
Just couldn't figure out how to pass the event object up!